python/src/pyarrow/adapters/pandas.cc

```diff
@@ -1648,7 +1649,17 @@ inline Status ArrowSerializer<TYPE>::ConvertTypedLists(
         RETURN_NOT_OK(value_builder->Append(data, size));
       }
     } else if (PyList_Check(objects[i])) {
-      return Status::TypeError("Python lists are not yet supported");
+      PyAcquireGIL lock;
```

Comment:
I'm not sure if we should maybe acquire the GIL always in this function instead of only for pylists.